### PR TITLE
feat: miner: WithdrawBalance API

### PR DESCRIPTION
**Pull request by LexLuthr**
## Related Issues

https://github.com/filecoin-project/lotus/issues/7367

## Proposed Changes

Create a new API to withdraw balance from miner Actor. Currently, it is done complete within the CLI code.

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [x] This PR has tests for new functionality or change in behaviour
- [x] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

File: node/impl/storminer.go
Comment (geoff-vball) [inline node/impl/storminer.go:1301]: I don't think this comment is relevant anymore

**Review by geoff-vball — COMMENTED** [review]
The implementation here looks good to me. I'd like someone else on the team to comment on if everything is in the right spot/api before merging this

File: node/impl/storminer.go
Comment (LexLuthr) [inline node/impl/storminer.go:1301]:
```suggestion
		AmountRequested: amount,
```

File: node/impl/storminer.go
Comment (magik6k) [inline node/impl/storminer.go:1331]: Can we leave this up to the user, and just instantly return `smsg.Cid()`?

Comment (LexLuthr) [reply node/impl/storminer.go:1331]: Maybe we can implement StateWaitMsg part in the cmd rather than the API. Would that make more sense?

File: api/api_storage.go
Comment (magik6k) [inline api/api_storage.go:54]: One final nitpick - can we call this method `ActorWithdrawBalance`, to keep it grouped with the other methods which are related to the on-chain actor?

Comment (LexLuthr) [reply api/api_storage.go:54]: Sure. I will rename it to ActorWithdrawBalance

Comment (LexLuthr) [reply api/api_storage.go]: @magik6k Done

**Review by magik6k — APPROVED** [review]

**Pull request merged**
